### DISCUSSION TITLE
About the new component format replacing NBT

### DISCUSSION THREAD

**Discussion by vico93** [cc-tweaked/CC-Tweaked#1787] [General]
Just for curiosity, how you plans to handle the transition to "structured components" from NBT that Mojang is making starting in [1.20.5](https://minecraft.wiki/w/Java_Edition_1.20.5)?

Would this impact CC:T in a overall positive or negative way?

**Comment by SquidDev** [comment]
It's a bit like most of the major overhauls Minecraft has undergone (1.8's blockstate/model changes, 1.13's ... everything). It's definitely a good long-term change, but also going to be a massive porting effort — CC:T is a small mod, but I'd estimate there's 50+ hours of work required to update — don't expect mods to be updated any time soon.

**Reply by vico93** [reply]
> don't expect mods to be updated any time soon

\*Sweating profusely after realizing this\*

> It's definitely a good long-term change, but also going to be a massive porting effort — CC:T is a small mod, but I'd estimate there's 50+ hours of work required to update

Good work with this port Squid! And thank you for everything you done by the community!
